fdbserver/TLogServer.actor.cpp

```diff
@@ -3129,6 +3138,15 @@ ACTOR Future<Void> tLogStart(TLogData* self, InitializeTLogRequest req, Locality
 	self->popOrder.push_back(recruited.id());
 	self->spillOrder.push_back(recruited.id());
 
+	TraceEvent ev("Role", recruited.id());
+	ev.detail("As", Role::TRANSACTION_LOG.roleName)
+		    .detail("Transition", "BeginWithEpochInfo")
+		    .detail("Origination", recovering ? "Recovered" : "Recruited")
+		    .detail("OnWorker", self->workerID)
+		    .detail("RecoveryCount", std::to_string(logData->recoveryCount))
+		    .detail("SharedTLog", self->dbgid.shortString());
+	ev.trackLatest(recruited.id().shortString() + ".Role");
+
```

Comment:
Have you considered integrating this log entry into the TLogStart below?

Reply:
Yes. I should.